### PR TITLE
[KBFS-2244] jenkins: Move change checking to the Jenkinsfile

### PR DESCRIPTION
**Pull request by jzila**
Along with keybase/jenkins-helpers#10, moves Go change checking to the Jenkinsfile. We can also do git diff checking on _any_ subdirectory including JS, but since that script currently works, I haven't bothered to move it yet.

Whitespace-agnostic reviewing is your friend :)

https://github.com/keybase/client/pull/7245/files?w=1

### CODE REVIEW

**Comment by jzila** [comment]
I only need one review, but PTAL.

https://ci.keyba.se/job/client/job/PR-7245/9/ ran quickly, then when I removed `go/test/jenkins_test.sh`, it runs slowly. Seems to work!

File: Jenkinsfile [hunk 1]
```diff
@@ -95,18 +95,25 @@ helpers.rootLinuxNode(env, {
             }
         }
 
+        def hasGoChanges = hasChanges('go')
+        def hasJSChanges = hasChanges('shared')
```
Comment (chrisnojima) [inline Jenkinsfile]: Just a heads up the reason why I did this checking in the shell script was so we could have finer grained control. Not just 'does this top level folder have changes'. This is fine just pointing that out

**Comment by jzila** [comment]
If my build passes, I'm going to merge without a review so CI speeds up for tomorrow; I'd appreciate a retroactive review anyway though.

File: Jenkinsfile [hunk 2]
```diff
@@ -119,16 +126,16 @@ helpers.rootLinuxNode(env, {
...
+                                        if (hasGoChanges) {
+                                            testNixGo("Linux")
+                                        }
                                     }},
                                     test_linux_js: { withEnv([
                                         "PATH=${env.HOME}/.node/bin:${env.PATH}",
                                         "NODE_PATH=${env.HOME}/.node/lib/node_modules:${env.NODE_PATH}",
                                     ]) {
                                         dir("shared") {
-                                            stage("JS tests") {
```
Comment (chrisnojima) [inline Jenkinsfile]: Why remove the stage directive? Helps see where things are in the pipeline viewer

File: Jenkinsfile (hunk 1 above)
Comment (jzila) [inline Jenkinsfile]: Yeah I haven't actually moved any of the JS checking yet, this is just output for now. The JS script has more functionality that isn't as critical to implement in the Jenkinsfile as we don't need its reuse.

File: Jenkinsfile (hunk 2 above)
Comment (jzila) [inline Jenkinsfile]: It doesn't actually do anything inside the `parallel` node...

**Comment by chrisnojima** [comment]
? Doesn't this add a label in the pipeline (not blue ocean)

**Comment by jzila** [comment]
Oh neat, you are correct. All right I'll roll back that particular change.

**Pull request merged**